COSORE Winter Rs Analysis.Rmd

```diff
@@ -112,45 +265,76 @@ for(i in 1:length(winterdata_names)){
 
 ft_totals<-bind_rows(ft_results, .id = "dataset")
 
-```
+# SOON: Update this code so it gives you the number of freeze thaw cycles in <each year> of <each dataset> that has <clean> data 
 
 
-# Pull out years that do not have enough data
+```
 
-# Calculate Annual Rs by Year
-Current def of year is Jan 1 to Dec 31. 
 
-```{r rs-by-year}
+# Calculate Annual Rs by Year
+Current definition of year is Jan 1 to Dec 31. 
 
-anRsresults <- list()
+```{r rs-by-year-fixed}
 
-for(i in 1:length(winterdata_names)){
-  message("I'm on ", winterdata_names[i]," now")
-  ds<-winterdata_full[[i]]$data
-  ds %>%
-    select(CSR_TIMESTAMP_BEGIN, CSR_FLUX_CO2) %>% 
-    filter( ! is.na(CSR_FLUX_CO2) ) %>% 
-    mutate(year= year(CSR_TIMESTAMP_BEGIN)) %>%
-    group_by(year) %>%
-    summarise(avgRs=mean(CSR_FLUX_CO2)) -> ds1
+winterdata_bound %>% 
+  group_by(dataset, year) %>% 
+  summarise(avgRs=mean(CSR_FLUX_CO2, na.rm= TRUE)) -> anRs_increment
   
-  ds1 %>% 
-    mutate(anRs=avgRs*31.536*12.0107) -> anRsresults[[i]]
-}
-
-anRs_totals<-bind_rows(anRsresults, .id = "dataset")
-#bug here, we want names of dataset
-#I don't know how reasonable these Rs numbers are
+anRs_increment %>% 
+  mutate(anRs=avgRs*31.536*12.0107) -> winterdata_annualRs
 
 ```
 
 # Calculate Winter Rs 
 Within calendar year Jan 1 - March 20, Dec 21 - Dec 31  
-```{}
+```{r winter-Rs}
+
+winterdata_bound %>% 
+  filter(winter == TRUE) %>% 
+  group_by(dataset, year) %>% 
+  summarise(avg_winter_Rs=mean(CSR_FLUX_CO2, na.rm= TRUE)) -> wRs_increment
+ 
+wRs_increment %>% 
+  mutate(winter_Rs=avg_winter_Rs*31.536*12.0107*0.25) -> winterdata_summary
```

Comment:
This line could use a comment explaining what these constants are and, in general, what the conversion/calculation being done here is..

Reply:
Yes, good point!